### PR TITLE
DO-300: Migrate & Re-factor code to move family member info in member…

### PR DESCRIPTION
**Pull request by sandeep-pepcus**
… info

### CODE REVIEW

File: src/main/java/com/pepcus/directoryonline/api/model/convertor/MemberModelConvertor.java [hunk 1]
```diff
@@ -262,11 +262,13 @@ private static SocialNetworkModel convertMemberEntityToSocialNetworkModelList(
...
-    return FamilyDetailModel.builder().id(familyDetail.getId()).name(familyDetail.getName())
-        .age(familyDetail.getAge()).dob(DateTimeUtils.getDate(familyDetail.getDob(), DATE_PATTERN))
+      Member familyDetail) {
+    return FamilyDetailModel.builder().id(familyDetail.getId()).firstName(familyDetail.getFirstName())
+        .middleName(familyDetail.getMiddleName()).lastName(familyDetail.getLastName()).phoneNumber(familyDetail.getPrimaryNumber())
+        .interest(familyDetail.getInterest()).imgeUrl(familyDetail.getImageUrl())
+        .age(familyDetail.getAge()).dob(DateTimeUtils.getDate(familyDetail.getDateOfBirth(), DATE_PATTERN))
         .relationshipType(familyDetail.getRelationship().name()).gender(familyDetail.getGender())
-        .education(familyDetail.getEducation()).profession(familyDetail.getProfession()).build();
+        .education(familyDetail.getEducation()).profession(familyDetail.getProfession()).imgeUrl(familyDetail.getImageUrl()).build();
```
Comment (pepcus-ankit) [inline src/main/java/com/pepcus/directoryonline/api/model/convertor/MemberModelConvertor.java:271]: imgeUrl -->> typo.

File: src/main/java/com/pepcus/directoryonline/api/model/FamilyDetailModel.java [hunk 2]
```diff
@@ -41,7 +47,15 @@
...
   private String profession;
+  
+  @JsonDeserialize(using = CustomDeserializer.class)
+  private String phoneNumber;
 
+  @JsonDeserialize(using = CustomDeserializer.class)
+  private String interest;
+  
+  @JsonDeserialize(using = CustomDeserializer.class)
+  private String imgeUrl;
```
Comment (pepcus-ankit) [inline src/main/java/com/pepcus/directoryonline/api/model/FamilyDetailModel.java:58]: imageUrl

Comment (pepcus-ankit) [reply src/main/java/com/pepcus/directoryonline/api/model/FamilyDetailModel.java:58]: Make sure to add entry for new attributes in default.json fieldLabels (if new attribute key added)

File: src/main/java/com/pepcus/directoryonline/api/service/MemberService.java [hunk 3]
```diff
@@ -264,4 +266,37 @@ public TenantMember getTenantMemberByMemberAndLoginId(Integer memberId, Integer
...
+   * 
+   * @param topFields
+   * @param memberList
+   * @return list of member object
+   */
+  public List<Member> getMemberList(List<String> topFields, List<Member> memberList) {
+    List<Member> listOfMemberObjects = new ArrayList<>();
+    memberList.stream().forEach(memberObj -> {
+      Member member = new Member();
+      topFields.stream().forEach(item -> {
```
Comment (pepcus-ankit) [inline src/main/java/com/pepcus/directoryonline/api/service/MemberService.java:281]: I was telling that we should extract this as separate method setMemberTopAttributes

In addition, we may need to return member relationship in future for family members. In that case there could be different toplist to be set for family member.

Comment (sandeep-pepcus) [reply src/main/java/com/pepcus/directoryonline/api/service/MemberService.java]: Addressed. Extracted in separate method.

File: src/main/java/com/pepcus/directoryonline/api/model/FamilyDetailModel.java (hunk 2 above)
Comment (sandeep-pepcus) [reply src/main/java/com/pepcus/directoryonline/api/model/FamilyDetailModel.java]: Addressed. Fixed typo and Verified that all attributes entry is already available in default.json fieldLables.

File: src/main/java/com/pepcus/directoryonline/api/model/convertor/MemberModelConvertor.java (hunk 1 above)
Comment (sandeep-pepcus) [reply src/main/java/com/pepcus/directoryonline/api/model/convertor/MemberModelConvertor.java]: Fixed.

**Comment by sandeep-pepcus** [comment]
Need to merge UI & Back-end PR simultaneously.

File: src/main/java/db/migration/V5__DO_300_Migrate_Family_Detail_To_Member.java [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            name = param.toString();
+          }
+          if ("age".equals(key)) {
+            age = (param != null) ? (Integer) param : null;
+          }
+          if ("dob".equals(key)) {
+            dob = (param != null) ? (Date) param : null;
+          }
+          if ("relationship".equals(key)) {
+            relationship = (param != null) ? param.toString() : null;
```
Comment (pepcus-ankit) [inline src/main/java/db/migration/V5__DO_300_Migrate_Family_Detail_To_Member.java:71]: relationship can't be null.

File: src/main/resources/db/migration/V4__DO-300_alter_table_member_info.sql [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+-- 1. Alter table core.member_info to add new column relationship
+-------------------------------------------------------------------------------------------------------
+
+alter table core.member_info add column relationship_type core.relationship;
+
+-------------------------------------------------------------------------------------------------------
+-- 2. Alter table core.member_info to add new column parent_id
+-------------------------------------------------------------------------------------------------------
+
+alter table core.member_info add column parent_id integer,
```
Comment (pepcus-ankit) [inline src/main/resources/db/migration/V4__DO-300_alter_table_member_info.sql:20]: why two alter script for altering same table?

File: src/main/java/db/migration/V5__DO_300_Migrate_Family_Detail_To_Member.java (hunk 4 above)
Comment (sandeep-pepcus) [reply src/main/java/db/migration/V5__DO_300_Migrate_Family_Detail_To_Member.java]: Resolved

File: src/main/resources/db/migration/V4__DO-300_alter_table_member_info.sql (hunk 5 above)
Comment (sandeep-pepcus) [reply src/main/resources/db/migration/V4__DO-300_alter_table_member_info.sql]: Modified.

**Pull request merged**